### PR TITLE
Refactoring `TimeSeriesTypeSource` and cleaning up `CsvFileConnector`

### LINKED ISSUE #565: Consolidate `TypeEntry` and `IndividualTimeSeriesMetaInformation`
With #546, `TimeSeriesTypeSource` is introduced, including the `TimeSeriesTypeFactory` which produces `TypeEntry`s.

This now leads to duplicates: `edu.ie3.datamodel.io.source.TimeSeriesTypeSource.TypeEntry` and `edu.ie3.datamodel.io.naming.timeseries.IndividualTimeSeriesMetaInformation` have the same contents. 

Furthermore, `CsvIndividualTimeSeriesMetaInformation` are built by `CsvFileConnector`, while the `SqlConnector` does not provide the `IndividualTimeSeriesMetaInformation`.

These descrepancies should be fixed.

cc @t-ober 

In order to re-allocate the responsibility to provide specific meta information and adapt to the needs of SIMONA, the following interface method should be re-introduced:

```
Optional<IndividualTimeSeriesMetaInformation> getTimeSeriesMetaInformation(UUID timeSeriesUuid);
```

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #565 and #544

- [x] Refactoring `TimeSeriesTypeSource` to `TimeSeriesMetaInformationSource`
- [x] Merging `IndividualTimeSeriesMetaInformation` and `TypeEntry`
- [x] Re-introducing `TimeSeriesMetaInformationSource#getTimeSeriesMetaInformation(UUID)`
- [x] Saving retrieved time series meta information in `CsvTimeSeriesMetaInformationSource` instead of `CsvFileConnector`
- [x] #544
- [x] Removing deprecated SQLs from test resources 
- [x] Adapting CHANGELOG

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/io/source/TimeSeriesMappingSource.java [hunk 1]
```diff
@@ -38,7 +38,8 @@ default Optional<UUID> getTimeSeriesUuid(UUID modelIdentifier) {
    *
    * @param timeSeriesUuid Unique identifier of the time series in question
    * @return An Option onto the meta information
-   * @deprecated since 3.0. Use {@link TimeSeriesTypeSource#getTimeSeriesMetaInformation()} instead
+   * @deprecated since 3.0. Use {@link
+   *     TimeSeriesMetaInformationSource#getTimeSeriesMetaInformation()} instead
    */
   @Deprecated(since = "3.0", forRemoval = true)
   Optional<edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation>
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/source/TimeSeriesMappingSource.java:45]: @sonatype-lift ignore

File: src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java [hunk 2]
```diff
@@ -221,87 +224,45 @@ public BufferedReader initReader(String filePath) throws FileNotFoundException {
...
    */
   @Deprecated(since = "3.0", forRemoval = true)
   public Optional<edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation>
       getIndividualTimeSeriesMetaInformation(UUID timeSeriesUuid) {
     if (Objects.isNull(individualTimeSeriesMetaInformation))
-      individualTimeSeriesMetaInformation = buildIndividualTimeSeriesMetaInformation();
+      individualTimeSeriesMetaInformation = getCsvIndividualTimeSeriesMetaInformation();
 
     return Optional.ofNullable(individualTimeSeriesMetaInformation.get(timeSeriesUuid))
         .map(edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation::new);
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java:237]: @sonatype-lift ignore

File: src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java [hunk 3]
```diff
@@ -221,87 +224,45 @@ public BufferedReader initReader(String filePath) throws FileNotFoundException {
    *
    * @param timeSeriesUuid The time series in question
    * @return An option on the queried information
-   * @deprecated since 3.0. Use {@link #getIndividualTimeSeriesMetaInformation()} instead
+   * @deprecated since 3.0. Use {@link #getCsvIndividualTimeSeriesMetaInformation(ColumnScheme...)}
+   *     instead
    */
   @Deprecated(since = "3.0", forRemoval = true)
   public Optional<edu.ie3.datamodel.io.csv.timeseries.IndividualTimeSeriesMetaInformation>
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java:231]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (danielfeismann) [reply src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java:231]: @sonatype-lift ignore

File: src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielfeismann) [reply src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java:237]: @sonatype-lift ignore

**Review by danielfeismann — APPROVED** [review]
Cool, thanks for the work. Looks good to me :)

**Pull request merged**